source/core/Console.cpp

```diff
@@ -0,0 +1,495 @@
+			{
+				RegisterCommand(static_cast<ConsoleCommand^>(attribute), method, defaultCommands);
+			}
+		}
+	}
+
+	void ConsoleScript::RegisterCommand(ConsoleCommand ^ command, System::Reflection::MethodInfo ^ methodInfo)
+	{
+		RegisterCommand(command, methodInfo, false);
+	}
+
+	void ConsoleScript::RegisterCommand(ConsoleCommand ^ command, System::Reflection::MethodInfo ^ methodInfo, bool defaultCommand)
+	{
+		File::WriteAllText("console.log", (command == nullptr) + ":" + (methodInfo == nullptr));
+		command->Name = defaultCommand ? methodInfo->Name : methodInfo->DeclaringType->FullName + "." + methodInfo->Name; //TODO FIX
+		command->Namespace = defaultCommand ? "Default Commands" : methodInfo->DeclaringType->FullName;
+		for each(auto args in methodInfo->GetParameters())
+		{
+			command->ConsoleArgs->Add(gcnew ConsoleArg(args->ParameterType->Name, args->Name));
+		}
+		if (!_commands->ContainsKey(command->Namespace))
+			_commands[command->Namespace] = gcnew List<Tuple<ConsoleCommand^, MethodInfo^>^>();
+
+		_commands[command->Namespace]->Add(gcnew Tuple<ConsoleCommand^, MethodInfo^>(command, methodInfo));
+		Info("Registered Command: " + command->Name);
+	}
+
+	void ConsoleScript::PrintHelpString()
+	{
+		StringBuilder ^help = gcnew StringBuilder();
+		help->AppendLine("--- Help ---");
+		for each(auto namespaceS in _commands->Keys)
+		{
+			help->AppendLine(String::Format("[{0}]", namespaceS));
+			for each(auto command in _commands[namespaceS])
+			{
+				auto consoleCommand = command->Item1;
+
+				help->AppendLine("    " + consoleCommand->BuildFormattedHelp());
+			}
+		}
+		Info(help->ToString());
+	}
+
+	void ConsoleScript::Clear()
+	{
+		_lines->Clear();
+	}
+
+	void ConsoleScript::OnTick(Object ^sender, EventArgs ^e)
+	{
+		DateTime now = DateTime::UtcNow;
+
+		array<String^> ^outputLines;
+		if (_outputQueue->TryDequeue(outputLines))
+		{
+			for each (String ^outputLine in outputLines)
+			{
+				_lines->AddFirst(outputLine);
+			}
+		}
+
+		if (_compilerTask != nullptr)
+		{
+			if (_compilerTask->IsCompleted)
+			{
+				Assembly ^compileResult = _compilerTask->Result;
+				if (compileResult != nullptr)
+				{
+					Type ^type = compileResult->GetType("ConsoleInput");
+					Object ^result = type->GetMethod("Execute")->Invoke(nullptr, nullptr);
+					if (result != nullptr)
+						Info(String::Format("[Return Value: {0}]", result));
+				}
+				ClearInput();
+				_compilerTask = nullptr;
+			}
+		}
+
+
+		//Hack so the input gets blocked long enogh
+		if (_lastClosed > now)
+		{
+			SetControlsEnabled(false);
+			return;
+		}
+
+		if (!_isOpen)
+			return;
+
+		SetControlsEnabled(false);
+
+		DrawRect(0, 0, WIDTH, HEIGHT / 3, BackgroundColor);
+		DrawRect(0, HEIGHT / 3, WIDTH, InputHeight, AltBackgroundColor);
+
+		DrawText(0, HEIGHT / 3, "$>", DefaultScale, DefaultFont, PrefixColor);
+		DrawText(20, HEIGHT / 3, _input, DefaultScale, DefaultFont, _compilerTask == nullptr ? InputColor : InputColorBusy);
```

Comment:
Increase the x value a bit here, so there is more space between the "$>" and command.